### PR TITLE
Arc 205 ros1 cleanup

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR merges the ROS1 `j2735_msgs` and `j3224_msgs` packages into `j2735_v2x_msgs` and `j3224_v2x_msgs`, which can be built with both ROS1 and ROS2.

## Related GitHub Issue

NA

## Related Jira Key

[ARC-205](https://usdot-carma.atlassian.net/browse/ARC-205)

## Motivation and Context

Consolidating these packages is a step towards our ROS2 migration efforts.

## How Has This Been Tested?

Messages built locally in ROS1 and ROS2

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[ARC-205]: https://usdot-carma.atlassian.net/browse/ARC-205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: carma_msgs/srv/Bind.srv [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# This is a service provided by each device driver to allow consumers
+# to inform the driver they request to bind with the given ID
+#
+
+# Request
+string id
+
+---
+
+# Response
```
Comment (MishkaMN) [inline carma_msgs/srv/Bind.srv:12]: Nit: Could you double check whether if we actually need this change? If it is part of the next round of changes should we wait for this one for PR's consistency?

File: j2735_v2x_msgs/CMakeLists.txt [hunk 2]
```diff
@@ -19,55 +19,97 @@ project(j2735_v2x_msgs)
...
-        ${msgs}
+find_package(ros_environment REQUIRED)
+
+set(ROS_VERSION $ENV{ROS_VERSION})
+
+if(${ROS_VERSION} EQUAL 1) # ROS 1
+  find_package(catkin REQUIRED
+    COMPONENTS
+    message_generation
+    std_msgs
```
Comment (MishkaMN) [inline j2735_v2x_msgs/CMakeLists.txt:32]: ROS1 Cmake file had geometry_msgs, is that actually not needed? same comment on `generate_messages` and `catkin_package`

File: j2735_v2x_msgs/CMakeLists.txt [hunk 3]
```diff
@@ -19,55 +19,97 @@ project(j2735_v2x_msgs)
...
+  )
+
+  file(GLOB msg_files msg/*.msg)
```
Comment (MishkaMN) [inline j2735_v2x_msgs/CMakeLists.txt:35]: This line is same for both ROS1 and 2, let's take it out of the if clauses from both.

File: j2735_v2x_msgs/CMakeLists.txt [hunk 4]
```diff
@@ -19,55 +19,97 @@ project(j2735_v2x_msgs)
...
+  )
 
-install(
-  FILES rule.yml
-  DESTINATION share/${PROJECT_NAME})
+  install(DIRECTORY include/${PROJECT_NAME}/
+    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN ".svn" EXCLUDE
+  )
```
Comment (MishkaMN) [inline j2735_v2x_msgs/CMakeLists.txt:67]: NIT: These lines look awfully similar to the one we have in ROS2. We can factor it out by 

1. replacing this here with

```
set(INCLUDE_DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
```
2. Similarly, take out the similar `install` block from ROS2, and replace with

```
set(INCLUDE_DESTINATION include/${PROJECT_NAME})
```
3. outside the if else block, add
```

# Common for both ROS 1 and ROS 2
install(DIRECTORY include/${PROJECT_NAME}/
  DESTINATION ${INCLUDE_DESTINATION}
  FILES_MATCHING PATTERN "*.h"
  PATTERN ".svn" EXCLUDE
)

```

File: j3224_v2x_msgs/CMakeLists.txt [hunk 5]
```diff
@@ -19,36 +19,82 @@ project(j3224_v2x_msgs)
...
 find_package(carma_cmake_common REQUIRED)
-carma_check_ros_version(2)
 carma_package()
 
-## Find ament macros and libraries
-find_package(rosidl_default_generators REQUIRED)
-find_package(ament_cmake REQUIRED)
-find_package(std_msgs REQUIRED)
-find_package(j2735_v2x_msgs REQUIRED)
+find_package(ros_environment REQUIRED)
```
Comment (MishkaMN) [inline j3224_v2x_msgs/CMakeLists.txt:24]: Similar comments with j2735

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: carma_msgs/srv/Bind.srv (hunk 1 above)
Comment (john-chrosniak) [reply carma_msgs/srv/Bind.srv:12]: This change will be needed in CAR-6069 when the mock lightbar driver is ported to ROS2, but yeah good point that this probably belongs in a separate PR. I'll go ahead and remove this

File: j2735_v2x_msgs/CMakeLists.txt (hunk 2 above)
Comment (john-chrosniak) [reply j2735_v2x_msgs/CMakeLists.txt:32]: Yes none of the messages in the package use `geometry_msgs`, so this package shouldn't be needed

File: j2735_v2x_msgs/CMakeLists.txt (hunk 3 above)
Comment (john-chrosniak) [reply j2735_v2x_msgs/CMakeLists.txt:35]: Good catch, I'll move that out and the first `set(msgs )` as well

File: j2735_v2x_msgs/CMakeLists.txt (hunk 4 above)
Comment (john-chrosniak) [reply j2735_v2x_msgs/CMakeLists.txt]: Great suggestion, that looks much cleaner

File: carma_msgs/srv/Bind.srv (hunk 1 above)
Comment (saina-ramyar) [reply carma_msgs/srv/Bind.srv]: FYI mock lightbar driver will be removed and not ported to ROS2.
https://usdot-carma.atlassian.net/browse/CAR-6069

Comment (john-chrosniak) [reply carma_msgs/srv/Bind.srv]: Got it, definitely makes sense to remove this then

File: j2735_v2x_msgs/CMakeLists.txt [hunk 6]
```diff
@@ -25,15 +25,16 @@ find_package(ros_environment REQUIRED)
 
 set(ROS_VERSION $ENV{ROS_VERSION})
 
+file(GLOB msg_files msg/*.msg)
+set(msgs )
```
Comment (MishkaMN) [inline j2735_v2x_msgs/CMakeLists.txt:29]: Nit: Sorry I missed this in the first round. I think we can remove `set(msgs )`, which I think is only there to declare the variable empty and we may not need to do that. Could you check if we can build it fine wihout it? I'll approve it though

File: j3224_v2x_msgs/CMakeLists.txt [hunk 7]
```diff
@@ -25,6 +25,9 @@ find_package(ros_environment REQUIRED)
 
 set(ROS_VERSION $ENV{ROS_VERSION})
 
+file(GLOB msg_files msg/*.msg)
+set(msgs )
```
Comment (MishkaMN) [inline j3224_v2x_msgs/CMakeLists.txt:29]: Ditto

**Review by MishkaMN — DISMISSED** [review]

**Review by MishkaMN — APPROVED** [review]

**Pull request merged**